Sources/BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -283,6 +285,10 @@ public class SendTokenStore: ObservableObject {
         addressError = .sameAsFromAddress
         return
       }
+      guard domain == sendAddress, !Task.isCancelled else {
+        // address changed while resolving, or validation cancelled.
+        return
```

Comment:
maybe we should disable input while the address is being resolved?

Reply:
If we disable input while the address is being resolved it would close the software keyboard on the user while they are typing. I don't think this would be good UX on mobile, even increasing the current throttle from 0.25s to a high value would be tricky to find a nice 'sweet spot' to not interfere with the user.

https://user-images.githubusercontent.com/5314553/224394842-022416b0-cddc-4e1d-9d91-4e46e2a93edb.mp4

With this change we ignore the first request result if the send address / domain changed while we were resolving it (the request for current send address / domain pending).